### PR TITLE
Add a 'How to resolve "Sourcegraph.com is deleted entirely"' incident handling guide

### PR DESCRIPTION
**Pull request by slimsag**
Fixes https://github.com/sourcegraph/sourcegraph/issues/10454

For any suggestions, please [make the edits directly](https://github.com/sourcegraph/about/edit/sg/dotcom-deletion/handbook/engineering/incident_dotcom_deleted_entirely.md?pr=%2Fsourcegraph%2Fabout%2Fpull%2F904)

### CODE REVIEW

File: handbook/engineering/incident_dotcom_deleted_entirely.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+- Navigate to the `sourcegraph-dev` project and [look at the existing Kubernetes clusters](https://console.cloud.google.com/kubernetes/list?project=sourcegraph-dev). Does the `dot-com` cluster exist still?
+  - No, the dot-com cluster is gone:
+    - Do the disks for the now-deleted cluster nodes still exist? Check by navigating to [Compute -> Disks](https://console.cloud.google.com/compute/disks?project=sourcegraph-dev) and searching for `pgsql-dot-com`
+      - Yes, the disks still exist: Go to **Recreating GKE cluster** and follow the **with existing disks** steps.
+      - No, the disks are gone: Go to **Recreating GKE cluster** and follow the **from snapshots** steps.
+  - Yes, the dot-com cluster exists: Go to **Recreating Kubernetes objects**
+
+## Recreating GKE cluster
+
+1. [Navigate to the `dot-com` cluster pn the Google Cloud console](https://console.cloud.google.com/kubernetes/list?project=sourcegraph-dev) and click `Connect`, run the `gcloud command it gives you.
```
Comment (uwedeportivo) [inline handbook/engineering/incident_dotcom_deleted_entirely.md:14]:
```suggestion
1. [Navigate to the `dot-com` cluster pn the Google Cloud console](https://console.cloud.google.com/kubernetes/list?project=sourcegraph-dev) and click `Connect`, run the gcloud command it gives you.
```

File: handbook/engineering/incident_dotcom_deleted_entirely.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+```
+gcloud beta container --project "sourcegraph-dev" clusters create "dot-com" --zone "us-central1-f" --no-enable-basic-auth --cluster-version "1.14.10-gke.27" --machine-type "n1-standard-64" --image-type "COS" --disk-type "pd-ssd" --disk-size "500" --local-ssd-count "1" --metadata disable-legacy-endpoints=true --scopes "https://www.googleapis.com/auth/devstorage.read_only","https://www.googleapis.com/auth/logging.write","https://www.googleapis.com/auth/monitoring","https://www.googleapis.com/auth/servicecontrol","https://www.googleapis.com/auth/service.management.readonly","https://www.googleapis.com/auth/trace.append" --num-nodes "5" --enable-stackdriver-kubernetes --no-enable-ip-alias --network "projects/sourcegraph-dev/global/networks/default" --no-enable-master-authorized-networks --addons HorizontalPodAutoscaling,HttpLoadBalancing --enable-autoupgrade --enable-autorepair --max-surge-upgrade 1 --max-unavailable-upgrade 0
+```
+3. If the above does not work, ensure you use the same properties (specifically "dot-com" cluster name, "us-central1-f zone, 5 of n1-standard-64 machine type, 500G boot disk size, 1 local SSD (default size).
```
Comment (uwedeportivo) [inline handbook/engineering/incident_dotcom_deleted_entirely.md:19]:
```suggestion
3. If the above does not work, ensure you use the same properties (specifically "dot-com" cluster name, "us-central1-f " zone, 5 of n1-standard-64 machine type, 500G boot disk size, 1 local SSD (default size).
```

File: handbook/engineering/incident_dotcom_deleted_entirely.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+- Navigate to the `sourcegraph-dev` project and [look at the existing Kubernetes clusters](https://console.cloud.google.com/kubernetes/list?project=sourcegraph-dev). Does the `dot-com` cluster exist still?
+  - No, the dot-com cluster is gone:
+    - Do the disks for the now-deleted cluster nodes still exist? Check by navigating to [Compute -> Disks](https://console.cloud.google.com/compute/disks?project=sourcegraph-dev) and searching for `pgsql-dot-com`
+      - Yes, the disks still exist: Go to **Recreating GKE cluster** and follow the **with existing disks** steps.
+      - No, the disks are gone: Go to **Recreating GKE cluster** and follow the **from snapshots** steps.
+  - Yes, the dot-com cluster exists: Go to **Recreating Kubernetes objects**
+
+## Recreating GKE cluster
+
+1. [Navigate to the `dot-com` cluster pn the Google Cloud console](https://console.cloud.google.com/kubernetes/list?project=sourcegraph-dev) and click `Connect`, run the gcloud command it gives you.
```
Comment (nicksnyder) [inline handbook/engineering/incident_dotcom_deleted_entirely.md:14]:
```suggestion
1. [Navigate to the `dot-com` cluster on the Google Cloud console](https://console.cloud.google.com/kubernetes/list?project=sourcegraph-dev) and click `Connect`, run the gcloud command it gives you.
```

File: handbook/engineering/incident_dotcom_deleted_entirely.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  b. Run `kubectl-apply-all.sh` and most things should come online but Sourcegraph.com will still be inaccessible.
+  c. Run [configure/ingress-nginx/install.sh](https://github.com/sourcegraph/deploy-sourcegraph-dot-com/blob/master/configure/ingress-nginx/install.sh) to install the nginx ingress.
+  d. Expose the cluster by running ONLY the `kubectl expose` commands found in [`create-new-cluster.sh`](https://github.com/sourcegraph/deploy-sourcegraph-dot-com/blob/release/create-new-cluster.sh#L61-L80)
+  e. Go to **Confirm health of Sourcegraph.com**
+6. **From snapshots**, recreate the Kubernetes objects:
+  a. Since nothing exists, run `create-new-cluster.sh` and it will recreate everything including disks.
+  b. Sourcegraph.com should now be accessible, but with no postgres, redis-store, or precise-code-intel-bundle-manager data present.
+  c. Restore pgsql disks from the latest `pgsql-prod-dot-com` compute snapshot, `redis-store-prod-dot-com` snapshot, and `bundle-manager-prod-dot-com` snapshot: https://console.cloud.google.com/compute/snapshots?project=sourcegraph-dev
+    - TODO: this section should be more explicit about what needs to be done.
```
Comment (daxmc99) [inline handbook/engineering/incident_dotcom_deleted_entirely.md:31]: Agreed, I would definitely struggle to know where to find these snapshots and the commands to run to restore them

File: handbook/engineering/incident_dotcom_deleted_entirely.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
+# How to resolve a "Sourcegraph.com is deleted entirely" incident
```
Comment (nicksnyder) [inline handbook/engineering/incident_dotcom_deleted_entirely.md:1]: True or false: ideally the content in the page would be "run command XYZ to ensure that sourcegraph.com is deployed correctly" (totally fine if the script is interactive and the dev has to make choices).

If true, can we file an issue to capture that work and then link to that issue from the top of this handbook page?

File: handbook/engineering/incident_dotcom_deleted_entirely.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+6. **From snapshots**, recreate the Kubernetes objects:
+  a. Since nothing exists, run `create-new-cluster.sh` and it will recreate everything including disks.
+  b. Sourcegraph.com should now be accessible, but with no postgres, redis-store, or precise-code-intel-bundle-manager data present.
+  c. Restore pgsql disks from the latest `pgsql-prod-dot-com` compute snapshot, `redis-store-prod-dot-com` snapshot, and `bundle-manager-prod-dot-com` snapshot: https://console.cloud.google.com/compute/snapshots?project=sourcegraph-dev
+    - TODO: this section should be more explicit about what needs to be done.
+  d. Go to **Confirm health of Sourcegraph.com**
+
+## Recreating Kubernetes objects
+
+1. [Navigate to the `dot-com` cluster pn the Google Cloud console](https://console.cloud.google.com/kubernetes/list?project=sourcegraph-dev) and click `Connect`, run the `gcloud command it gives you.
```
Comment (nicksnyder) [inline handbook/engineering/incident_dotcom_deleted_entirely.md:36]:
```suggestion
1. [Navigate to the `dot-com` cluster on the Google Cloud console](https://console.cloud.google.com/kubernetes/list?project=sourcegraph-dev) and click `Connect`, run the `gcloud command it gives you.
```

File: handbook/engineering/incident_dotcom_deleted_entirely.md (hunk 5 above)
Comment (slimsag) [reply handbook/engineering/incident_dotcom_deleted_entirely.md:1]: True and false.

In a perfect world, yes, that would be ideal. In the real world, I think it is always likely to require at least some of the following:

- A few different commands
- Inspecting the current state in some way, and running different commands based on what you see.
- Having your `gcloud`, `kubectl`, `terraform` commands installed and set up to point at the right projects, etc
- Explaining to someone who is not familiar with our tooling, repository structure, infrastructure, etc. what these magical commands are doing in a critical time without saying "here are 5 page-long readings, go follow those first"

Thus, I imagine this document getting easier and more clear over time -- but likely not ever going away entirely. https://github.com/sourcegraph/sourcegraph/issues/10455 is one such example of how this document will become easier but not go away.

File: handbook/engineering/incident_dotcom_deleted_entirely.md (hunk 4 above)
Comment (slimsag) [reply handbook/engineering/incident_dotcom_deleted_entirely.md:31]: @ggilmore can you comment on this and/or look into fleshing this section out more? I wasn't able to myself without spending significant time

**Comment by slimsag** [comment]
If anyone has further suggestions, feel free to send PRs! Merging as-is now because this is better than the status-quo.

**Pull request merged**